### ISSUE TITLE
Ignore div parents without landmark roles when determining landmark nesting

### ISSUE THREAD

**Issue by carmacleod** [matatk/landmarks#108]
In the following landmarks list:
![landmarks on page](https://user-images.githubusercontent.com/3331913/31355582-e66caf4a-ad08-11e7-8da5-2b95c3fca896.png)

Both "Active Branch (master)" and "Working Directory Changes" regions are great-great grandchildren of main. Yet they are shown at different levels in the landmarks list. I would expect them to be at the same level? The markup looks something like this:

```
<main>
    <div id="leftPane" class="sidePanelLayout hasSplit">
        <div class="fixedToolbarHolder sidebarWrapper">
            <div id="pageSidebar" class="toolbarTarget sidebar">
                <div role="region" aria-labelledby="branchTitle"></div>
            </div>
        </div>
    </div>
    <div class="split splitLayout"></div>
    <div id="rightPane" class="mainPanelLayout hasSplit">
        <div class="fixedToolbarHolder">
            <div id="table" class="workingTarget">
                <div role="region" aria-labelledby="workingDirChangesTitle"></div>
            </div>
        </div>
    </div>
</main>
````

**Comment by carmacleod** [comment]
Similarly, on this other page, all of these:
- README.md
- minesweeper-objects
- Project Information
- Git
- JavaScript

are children of nested divs in the Work Area region, but only the first one (README.md) is shown within the Work Area region, and for some reason the rest are shown as siblings of the Work Area region (which they aren't).

![image](https://user-images.githubusercontent.com/3331913/31360047-93e40374-ad1a-11e7-9db6-02baf0a6cd0d.png)

**Comment by matatk** [comment]
Thanks for the problem report and the test code. I may need to add a test case for this. Are you still using 2.1.0alpha1?

**Comment by carmacleod** [comment]
Yes, I'm using 2.1.0alpha on FF.
(Chrome version 2.0.8 doesn't find my dynamically added landmarks or labels).
PS: I'm giving a demo to some colleagues tomorrow, on how your extension helped me add landmarks to our product. It would be cool to show them on Chrome, because that's what most of them use... ;)
No pressure, though - I can show them just as well on FF. :)

**Comment by matatk** [comment]
I've made a slightly updated version of 2.1.0 for Chrome (it's still alpha, but on Chrome the version string can't contain letters, so I took that part off; nothing else has changed).  If you download it from here, and then unzip it and [use Google's instructions on loading local add-ons](https://developer.chrome.com/extensions/getstarted#unpacked) then you can load it into Chrome.  Note this has not had any checks/signing done to it by Google, but I've tested it :-).

I won't be able to release 2.1.0 proper for several weeks probably, as it's such a big change I need time to get a feel for how it's going.

[landmarks-2.1.0-chrome.zip](https://github.com/matatk/landmarks/files/1373343/landmarks-2.1.0-chrome.zip)

**Comment by carmacleod** [comment]
Thanks for Chrome 2.1.0! Much appreciated. Demo went well - I used both Chrome and FF, because the next landmark shortcut works in FF, but not Chrome. (I think you already knew that).

In case it's helpful, I had to edit the manifest.json file after unzipping Chrome 2.1.0 because:

> Failed to load extension from: ~\Downloads\landmarks-2.1.0-chrome
> Required value 'content_scripts[0].matches' is missing or invalid.

I just guessed, and added the following to the beginning of the content_scripts object, and it worked.  :)
```
      "matches": [
        "<all_urls>"
      ],
```

**Comment by matatk** [comment]
Oops; I did try running it here and didn't notice that error; sorry. Glad you were able to get it working :-).

**Comment by carmacleod** [comment]
No worries - it was great that you sent it.  :)

**Comment by matatk** [comment]
You were spot-on with the problem, and you've also discovered an ancient bug, which I believe is now fixed (it'll be in 2.1.0 when it's released); thanks :-).
